src/utils/runner.ts

```diff
@@ -3,53 +3,119 @@ import { TestRunRequest, TestResult } from "../models/types";
 import { cloneRepository, cleanupRepository } from "./gitUtils";
 import { reportResults } from "./resultReporter";
 import { buildDockerImage, runInDocker, cleanupDocker } from "./dockerUtils";
-// import fs from "fs/promises";
-// import path from "path";
+import { loadTestFile } from "./testLoader";
+import { fetchUserProgress } from "./progress";
 import { EVENT_TYPE } from "./constants";
 import logger from "./logger";
+import fs from "fs/promises";
+import path from "path";
+import { ProgressResponse } from "../models/types";
 
 export async function runTestProcess(request: TestRunRequest): Promise<void> {
   const { repoUrl, branch, commitSha } = request;
+
   let repoDir: string | null = null;
   let imageName: string | null = null;
 
   try {
+    // Fetch user's progress
+    const progressData: ProgressResponse[] = await fetchUserProgress(repoUrl);
```

Comment:
Progress data is an object not an array.

Reply:
oh, yea, havent rollout new change in k8s since the last update was pushed. will change now.

Reply:
fixed